lib/pkgr/builder.rb

```diff
@@ -284,17 +287,23 @@ def distribution
       @distribution ||= Distributions.current(config)
     end
 
-    # List of available buildpacks for the current distribution.
-    def buildpacks
-      distribution.buildpacks
-    end
-
-    # Buildpack detected for the app, if any.
-    def buildpack_for_app
+    # Buildpacks detected for the app, if any. If multiple buildpacks are explicitly specified, all are used
+    def buildpacks_for_app
       raise "#{source_dir} does not exist" unless File.directory?(source_dir)
-      @buildpack_for_app ||= buildpacks.find do |buildpack|
-        buildpack.setup(config.edge, config.home)
-        buildpack.detect(source_dir)
+      @buildpacks_for_app ||= begin
+        mode, buildpacks = distribution.buildpacks
+        case mode
+        when :custom
+          buildpacks.find_all do |buildpack|
+            buildpack.setup(config.edge, config.home)
+            buildpack.detect(source_dir)
+          end
+        else
+          [buildpacks.find do |buildpack|
+            buildpack.setup(config.edge, config.home)
+            buildpack.detect(source_dir)
+          end].compact
```

Comment:
@chytreg The behaviour would be exactly the same on Heroku. If we are going to have a *different* behaviour to Heroku, that's a big deal, and would be surprising for the majority of people who have used buildpacks before but never used pkgr.

What you refer to as "primitive" (without getting into a discussion on that itself) is what the vast majority of the users of any buildpack-based system are already used to.

Your proposal might be less surprising to you, but would be a shock to everyone who has prior experience of buildpacks. If we are going to do something that would shock those people, we should do so for a very good reason and document it thoroughly. I also believe that it should not form part of this PR, which is already good to go with no API changes. If your proposal is to go ahead, it should feature in a follow-up PR.